Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Scripts/InputSystemManager.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+#if UNITY_EDITOR
+using Microsoft.MixedReality.Toolkit.Input.Editor;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Service manager supporting running the input system, without requring the MixedRealityToolkit object.
+    /// </summary>
+    public class InputSystemManager : BaseServiceManager
+    {
+        [SerializeField]
+        [Tooltip("The input system type that will be instantiated.")]
+        [Implements(typeof(IMixedRealityInputSystem), TypeGrouping.ByNamespaceFlat)]
+        private SystemType InputSystemType = null;
+
+        [SerializeField]
+        [Tooltip("The input system configuration profile.")]
+        private MixedRealityInputSystemProfile profile = null;
+
+        private void Awake()
+        {
+            InitializeManager();
+        }
+
+        protected override void OnDestroy()
+        {
+            UninitializeManager();
+            base.OnDestroy();
+        }
+
+        /// <inheritdoc />
+        public override bool RegisterDataProvider<T>(T dataProviderInstance)
+        {
+            bool registered = base.RegisterDataProvider<T>(dataProviderInstance);
+            if (registered)
+            {
+                dataProviderInstance.Initialize();
+            }
+            return registered;
+        }
+
+        /// <summary>
+        /// Initialize the manager.
+        /// </summary>
+        private void InitializeManager()
+        {
+#if UNITY_EDITOR
+            // Make sure unity axis mappings are set.
+            InputMappingAxisUtility.CheckUnityInputManagerMappings(ControllerMappingLibrary.UnityInputManagerAxes);
+#endif
+
+            // The input system class takes arguments for:
+            // * The registrar
+            // * The input system profile
+            object[] args = { this, profile };
+            Initialize<IMixedRealityInputSystem>(InputSystemType.Type, args: args);
+
+            // The input system uses the focus provider specified in the profile.
+            // The args for the focus provider are:
+            // * The registrar
+            // * The input system profile
+            args = new object[] { this, profile };
+            Initialize<IMixedRealityFocusProvider>(profile.FocusProviderType.Type, args: args);
+
+            // The input system uses the raycast provider specified in the profile.
+            // The args for the focus provider are:
+            // * The registrar
+            // * The input system profile
+            args = new object[] { this, profile };
+            Initialize<IMixedRealityRaycastProvider>(profile.RaycastProviderType, args: args);
+
```

Comment:
It might be worth adding a comment stating that these lines are order dependent (I assume that the focus provider/raycast provider require the input system already having been registered?)